### PR TITLE
Graphite: Migrate to React (part 4 & 5: group all components)

### PR DESCRIPTION
**Pull request by ifrost**
**What this PR does / why we need it**:

Follow-up on #37309 and plan described in #36481. In this PR AngularJS template is limited to rendering a single React component:

Changes:
* All Graphite components are grouped in `GraphiteQueryEditor` and that's the only component rendered by Angular
* Moved Influx's `Section` component to grafana-ui to reuse it in `GraphiteQueryEditor` (updated storybook as well) (cc @grafana/grafana-frontend-platform)
* Created context for dispatch function to avoid passing it down
* Moved all React components to `components/` folder

**Which issue(s) this PR fixes**:

Fixes N/A

**Special notes for your reviewer**:

There are no changes to functionality. A lot of files changed but it's just reorganizing the code. Depends on #37309.

### CODE REVIEW

File: public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/Editor.tsx [hunk 1]
```diff
@@ -112,7 +103,7 @@ export const Editor = (props: Props): JSX.Element => {
 
   return (
     <div>
-      <SectionWrap initialName="from">
+      <Section label="FROM" fill={true}>
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/Editor.tsx:106]: Out of curiosity - is there a reason why we are changing lowercased labels to uppercased? This doesn't seem related to Graphite migration, therefore I would expect just refactor, not changed casing.

**Review by ivanahuckova — COMMENTED** [review]
From manual testing:
```
Adding functions
- open raw editor and type: seriesByTags('dc=~.*u.*', 'name=servers.requests.count')
- switch to visual editor and verify tags were added correctly, no function should be added (seriesByTag is a hidden function used when tags are in use)
```
In this case `seriesByTag` was actually added as function:
image


Otherwise looks and works great. 🚀 Not sure if this one issue needs to be fixed?

File: public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/Editor.tsx (hunk 1 above)
Comment (ifrost) [reply public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/Editor.tsx:106]: Influx has these labels in uppercase and the uppercase was applied by default in `SectionWrap` (this component part of Influx code). I moved `SectionWrap` to grafana-ui and called in `Section` to make it more generic, but making labels uppercase by default didn't look right to me. I changed the default behavior of the component to use the label as passed in label property. However, Influx editor displays labels in uppercase (it matches SQL syntax better) so I had to change it.

Comment (ivanahuckova) [reply public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/Editor.tsx:106]: Thanks for context! 🧡

**Comment by ifrost** [comment]
> From manual testing:
> 
> ```
> Adding functions
> - open raw editor and type: seriesByTags('dc=~.*u.*', 'name=servers.requests.count')
> - switch to visual editor and verify tags were added correctly, no function should be added (seriesByTag is a hidden function used when tags are in use)
> ```
> 
> In this case `seriesByTag` was actually added as function:
> image
> 
> Otherwise looks and works great. 🚀 Not sure if this one issue needs to be fixed?

Great catch! I forgot to map `ng-hide="func.hidden"` from:

```
  <div ng-repeat="func in ctrl.state.queryModel.functions" class="gf-form">
    <graphite-function-editor func="func" dispatch="ctrl.dispatch" ng-hide="func.hidden"></graphite-function-editor>
  </div>
```

**Review by ivanahuckova — APPROVED** [review]
LGTM! Nice work. 🤩 And thanks again for making testing plan - it makes manual testing so much smoother!

File: packages/grafana-ui/src/components/Section/SectionLabel.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { cx } from '@emotion/css';
+
+/**
+ * @alpha
+ */
+export const SectionLabel = ({ name, className }: { name: string; className?: string }) => (
+  <label className={cx('gf-form-label query-keyword', className)}>{name}</label>
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/Section/SectionLabel.tsx:8]: Is it possible to add a `for`attribute here?

File: packages/grafana-ui/src/components/Section/Section.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { SectionFill } from './SectionFill';
+import { SectionLabel } from './SectionLabel';
+
+/**
+ * @alpha
+ */
+export const Section = ({
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/Section/Section.tsx:8]: What's the purpose of this component? Does it only fit in the context of Segment? Is the label suppose to refer to something?

File: packages/grafana-ui/src/components/Section/Section.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  label,
+  children,
+  fill,
+}: {
+  label: string;
+  children: React.ReactNode;
+  inline?: boolean;
+  fill?: boolean;
+}) => (
+  <div className="gf-form-inline">
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/Section/Section.tsx:18]: If possible, use Emotion instead. I think `getFormStyles` might be of some help. The styles might be incompatible, so maybe that will have to be the next iteration.

**Comment by tskarhed** [comment]
Do you know if this is a side effect of this PR? 
Screenshot 2021-08-16 at 11 09 52

File: packages/grafana-ui/src/components/Section/SectionLabel.tsx (hunk 2 above)
Comment (ifrost) [reply packages/grafana-ui/src/components/Section/SectionLabel.tsx:8]: There's no single element it could point to. The label is for a row that contains multiple components, e.g. (stats, response, 500 are all separate segments):

Screen Shot 2021-08-16 at 12 22 22

File: packages/grafana-ui/src/components/Section/Section.tsx (hunk 3 above)
Comment (ifrost) [reply packages/grafana-ui/src/components/Section/Section.tsx:8]: It's a shortcut for creating horizontal sections / rows that may end with a filled space at the end (examples below). It can be used with any components but it's used with Segments at the moment. I'll move it `Segment` and rename to `SegmentsSections` to make it more explicit.

Screen Shot 2021-08-16 at 12 25 02
Screen Shot 2021-08-16 at 12 25 10
Screen Shot 2021-08-16 at 12 25 17

TODO:
- [ ] move Section to Segment

File: packages/grafana-ui/src/components/Section/SectionLabel.tsx (hunk 2 above)
Comment (tskarhed) [reply packages/grafana-ui/src/components/Section/SectionLabel.tsx:8]: Okay, in the next iteration it might be worth thinking about from an a11y point of view @Ijin08

File: packages/grafana-ui/src/components/Section/Section.tsx (hunk 3 above)
Comment (Clarity-89) [reply packages/grafana-ui/src/components/Section/Section.tsx:8]: We have `InlineField`, `inlineFieldRow` and `InlineLabel` components in grafana/ui, which should be able to achieve the same results. Or do they not work for this case?

File: packages/grafana-ui/src/components/Section/Section.tsx (hunk 4 above)
Comment (ifrost) [reply packages/grafana-ui/src/components/Section/Section.tsx]: Updated in 9822fed23ebdc760602de8aecedf9b50abdf03e9. It's using `useStyles2`. I couldn't use any of `getFormStyles` styles - shall I create a new entry there?

**Comment by ifrost** [comment]
> Do you know if this is a side effect of this PR?
> Screenshot 2021-08-16 at 11 09 52

Good spot! Should be fixed with 9822fed23ebdc760602de8aecedf9b50abdf03e9 (children were rendered in an incorrect place)

File: packages/grafana-ui/src/components/Section/Section.tsx (hunk 3 above)
Comment (ifrost) [reply packages/grafana-ui/src/components/Section/Section.tsx]: There a few differences:
- `SegmentSection` has fixed label styling (color and size) for consistency across editors
- `InlineField` takes only one child component, `SegmentSection` wraps multiple components

`SegmentSection` above vs `InflineField` below:

Screen Shot 2021-08-16 at 14 44 55

Comment (Clarity-89) [reply packages/grafana-ui/src/components/Section/Section.tsx]: What about wrapping multiple components into Fragment and passing them to `InlineLabel`? It just seems like these new components are 90% identical to the inline fields we already have, so I wonder if some customisation of the current component would make them work here?

Comment (ifrost) [reply packages/grafana-ui/src/components/Section/Section.tsx]: Make sense. I replaced them with `InlineFieldRow` and `InlineLabel` in 53901b0a0738aed163700fbfe85e675ccf0bc594. I hope it looks better now.

File: packages/grafana-ui/src/components/Segment/SegmentSection.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  return (
+    <>
+      <InlineFieldRow>
+        <InlineLabel width={12} className={styles.label}>
+          {label}
+        </InlineLabel>
+        {children}
+        {fill && (
+          <div className={styles.fill}>
+            <InlineLabel> </InlineLabel>
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/Segment/SegmentSection.tsx:35]:
```suggestion
            <InlineLabel />
```

**Review by Clarity-89 — APPROVED** [review]
Looks good, great work!

File: packages/grafana-ui/src/components/Segment/SegmentSection.tsx (hunk 5 above)
Comment (ifrost) [reply packages/grafana-ui/src/components/Segment/SegmentSection.tsx]: Children are required. Passing empty string in 8412395327cba9f2df05355dea2d18fbfab7ffeb now, or should I change children to be optional?
